editors/sc-ide/core/doc_manager.cpp

```diff
@@ -951,6 +955,26 @@ void DocumentManager::handleCloseDocScRequest(const QString& data) {
     }
 }
 
+void DocumentManager::handleSaveDocScRequest(const QString& data) {
+    try {
+        YAML::Node doc = YAML::Load(data.toStdString());
+        if (doc) {
+            if (!doc.IsSequence())
+                return;
+
+            std::string id = doc[0].as<std::string>();
+            std::string path = doc[1].as<std::string>();
+            Document* document = documentForId(id.c_str());
+            if (document) {
+                saveAs(document, QString::fromUtf8(path.c_str()));
+            }
+        }
```

Comment:
the code in this file could really do for some refactoring. i also see the following issues --

- no warnings / feedback given for any of the implicit `else` branches of if's
- possible unnecessary string copy
- no feedback given to language on completion

i don't expect you to solve any of that in this PR. after this is merged i wouldn't mind submitting one to work down this technical debt